src/top/cmd.rs

```diff
@@ -64,7 +64,8 @@ pub async fn cmd(opts: &super::Opts) -> exitcode::ExitCode {
 
     // Initialize the dashboard
     match init_dashboard(url.as_str(), opts, sender).await {
-        Ok(_) => exitcode::OK,
+        // Exit the process to clean up any lingering subscriptions.
+        Ok(_) => std::process::exit(0),
```

Comment:
This feels a bit weird to me in that the other match arm here returns the exit code rather than terminating the process directly. Is there a reason you needed to do this here rather than having the caller handle the exit code?

Reply:
The reason for the hard shut down at this point is to work around an active tungstenite client that's attempting to fire queries against a dead Websocket.

Even though the exit code is returned, since the thread is active, it prevents the process from quitting.

In #8770, I intend to introduce more granular control into the GraphQL subscription client for handling re-connection and ad hoc shutdown, which would make this block more elegant and allow it to propagate an exit code with any active subscriptions already cleaned up.

Another alternative would be to hard exit _after_ the sub-command block, rather than propagate the code back up as an `Err` (even if it's just `exitcode::OK` that's returned), but that seemed like a more invasive change rather than something that was specifically scoped to `top`.

For that reason, I elected to shutdown in the `top` command itself. Open to changing that per preference, however.

Reply:
Ah, yeah, I see. I'm fine with deferring the clean-up until we address https://github.com/vectordotdev/vector/issues/8770 . Would you want to just drop a link to that with a comment about cleaning up the subscriptions? I think that would make it less confusing for future readers who might have the same question I did.

Reply:
👍🏻 Added the following comment in 7be37e6754e9eceb48f45a5f6952ac91d5a98852:

> // Even though we could return an `exitcode::OK` here, the upstream block_on won't
> // return due to lingering subscriptions, which manifests as a hung cursor in the
> // terminal. To work around this for now, exit immediately.
> // TODO: https://github.com/vectordotdev/vector/issues/9206